SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -214,6 +217,25 @@ TArray<Worker_ComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 		ComponentDatas.Add(AuthorityIntent::CreateAuthorityIntentData(NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId()));
 	}
 
+	if (NetDriver->SpatialDebugger != nullptr)
+	{
+		if (SpatialSettings->bEnableUnrealLoadBalancer)
+		{
+			check(NetDriver->VirtualWorkerTranslator != nullptr);
+
+			VirtualWorkerId IntentVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+
+			const PhysicalWorkerName* PhysicalWorkerName = NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(IntentVirtualWorkerId);
+			FColor InvalidServerTintColor = NetDriver->SpatialDebugger->InvalidServerTintColor;
+			FColor IntentColor = PhysicalWorkerName == nullptr ? InvalidServerTintColor : SpatialGDK::GetColorForWorkerName(*PhysicalWorkerName);
```

Comment:
Not having your own physical worker name assigned to a virtual worker ID is a checkable issue imo?

Reply:
A few of the startup Actors are created before the Translator is ready (in particular WorldSettings) so I think we do need this check.